docs: update typescript.md (#38717)
vercel/next.js: docs/basic-features/typescript.md

You're now ready to start converting files from `.js` to `.tsx` and leveraging the benefits of TypeScript!

> > A file named `next-env.d.ts` will be created in the root of your project. This file ensures Next.js types are picked up by the TypeScript compiler. **You SHOULD NOT remove it or edit it** as it can change at any time. As such this file should not be committed and should be ignored by version control.

Reviewer: ```suggestion
> A file named `next-env.d.ts` will be created at the root of your project. This file ensures Next.js types are picked up by the TypeScript compiler. **You should not remove it or edit it** as it can change at any time. This file should not be committed and should be ignored by version control (e.g. inside your `.gitignore` file).
```
